### PR TITLE
test: add CI job to run multicluster demo

### PR DESCRIPTION
**Pull request by allenlsy**
[HOLD] pending on AKS clusters setup

Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

Add CI job to run bookbuyer -> bookstore test in multicluster mode.

There are several repository secrets need to be configured:

* `CTR_REGISTRY`: docker image registry used by the job
* `CTR_REGISTRY_USER`: registry username
* `CTR_REGISTRY_PASSWORD`:  registry password
* `AKS_MULTICLUSTER_KUBECONFIG`: kubeconfig file content, which configures cluster `alpha` and `beta`. 

Two clusters, alpha & beta, will be created on AKS. CI job will run the multicluster demo, send requests to verify functionalities.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [X] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

**Testing done**:

Tested with private AKS clusters. CI passed on forked repo.

image

image

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -305,6 +305,63 @@ jobs:
...
+        uses: azure/k8s-set-context@v1
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.AKS_MULTICLUSTER_KUBECONFIG }}
+        id: setcontext
+
+      - name: Login to ACR
+        uses: docker/login-action@v1
+        with:
+          registry: allenleigh.azurecr.io
```
Comment (snehachhabria) [inline .github/workflows/main.yml:346]: could this be a registry that is common for the team to use ?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -305,6 +305,63 @@ jobs:
...
+          username: allenleigh
+          password: ${{ secrets.CTR_REGISTRY_PASSWORD }}
+
+      - name: Run multicluster demo
+        env:
+          MULTICLUSTER_CONTEXTS: "alpha beta"
+          ALPHA_CLUSTER: alpha
+          BETA_CLUSTER: beta
+          MULTICLUSTER_MODE: true
```
Comment (snehachhabria) [inline .github/workflows/main.yml:355]: You will need to add BOOKSTORE_SVC=bookstore-v1 here

File: ci/cmd/maestro.go [hunk 3]
```diff
@@ -143,42 +207,60 @@ func maxWaitForOK() time.Duration {
 	return time.Duration(maxWaitInt) * time.Second
 }
 
-func getPodNames(kubeClient kubernetes.Interface) (string, string, string, string) {
+func getPodNames(kubeClient kubernetes.Interface, includeBooktheif, includeBookbuyer, includeBookstoreV1, includeBookstoreV2, includeBookwarehouse bool) (string, string, string, string) {
```
Comment (snehachhabria) [inline ci/cmd/maestro.go:210]:
```suggestion
func getPodNames(kubeClient kubernetes.Interface, includeBookthief, includeBookbuyer, includeBookstoreV1, includeBookstoreV2, includeBookwarehouse bool) (string, string, string, string) {
```

**Comment by allenlsy** [comment]
Nightly job is tested on forked repo. 

https://github.com/allenlsy/osm/actions/runs/1164703790

**Comment by snehachhabria** [comment]
stale PR, closing

**Pull request closed without merging**